pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go

```diff
@@ -213,7 +213,7 @@ func semanticHPAEquals(desired, existing *autoscalingv2.HorizontalPodAutoscaler)
 
 func shouldDeleteHPA(desired *autoscalingv2.HorizontalPodAutoscaler) bool {
 	desiredAutoscalerClass, hasDesiredAutoscalerClass := desired.Annotations[constants.AutoscalerClass]
-	return hasDesiredAutoscalerClass && constants.AutoscalerClassType(desiredAutoscalerClass) == constants.AutoscalerClassExternal
+	return false
```

Comment:
@terrytangyuan I think this is your original logic, was there any reason to delete HPA if it is set to external ?

Reply:
If this func `shouldDeleteHPA` is not needed, just remove it. The logic is not correct and not necessary.

Reply:
Thanks @houshengbo  for your suggestion.
Yes,  seems can remove this function in this user case.
However, not sure kserve may have deleteHPA requirements with other user cases later.
So, for HPAReconciler class general function (createHPA, updateHPA, deleteHPA), it may make sense be kept. Just FYI